### DISCUSSION TITLE
Nullable with generic types behaves so weird

### DISCUSSION THREAD

**Discussion by Cyl18** [dotnet/csharplang#4123] [General] [answered]
For example, I wrote a method that can get a list's element with index, and returns `default` when the index is above the list's length.
![image](https://user-images.githubusercontent.com/14993992/98817787-a1d01e80-2465-11eb-942d-1ecfa8c4b4e1.png)
`TValue` can be null, so I added a `?` to `TValue`
![image](https://user-images.githubusercontent.com/14993992/98817867-bf04ed00-2465-11eb-812d-f05603c7f108.png)
But this can't be done.  It suggests me to add a `class` or a `struct`, so I write this method separately:
![image](https://user-images.githubusercontent.com/14993992/98818195-2ae75580-2466-11eb-94ac-52b9e4a349b5.png)
And this can't be done too...

So I'm wondering if there is a way to achieve my thought?
*(um I'm a student and my English isn't quite good, so if there's anything that I didn't express clearly, please point it out!)*

**Comment by YairHalberstadt** [comment, marked as answer]
In C# 9 the first thing you wrote works.

**Reply by Cyl18** [reply]
I'd already changed to .NET 5 and C# 9, and it still showed that error, but it turns out that it's actually resharper's problem, after I turned off resharper, it works!
![image](https://user-images.githubusercontent.com/14993992/98819638-10ae7700-2468-11eb-9f30-586e565e18a0.png)
thank you!
